instrumentation/mongo/mongo-async-3.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/mongoasync/v3_3/MongoAsyncClientTest.java

```diff
@@ -0,0 +1,250 @@
+  }
+
+  @Override
+  protected InstrumentationExtension testing() {
+    return testing;
+  }
+
+  @Override
+  public void createCollection(String dbName, String collectionName) {
+    MongoDatabase db = client.getDatabase(dbName);
+    db.createCollection(collectionName, toCallback(result -> {}));
+  }
+
+  @Override
+  public void createCollectionNoDescription(String dbName, String collectionName) {
+    MongoDatabase db = MongoClients.create("mongodb://" + host + ":" + port).getDatabase(dbName);
+    db.createCollection(collectionName, toCallback(result -> {}));
+  }
+
+  @Override
+  public void createCollectionWithAlreadyBuiltClientOptions(String dbName, String collectionName) {
+    MongoClientSettings clientSettings = client.getSettings();
+    MongoClientSettings newClientSettings = MongoClientSettings.builder(clientSettings).build();
+    MongoDatabase db = MongoClients.create(newClientSettings).getDatabase(dbName);
+    db.createCollection(collectionName, toCallback(result -> {}));
+  }
+
+  @Override
+  public void createCollectionCallingBuildTwice(String dbName, String collectionName) {
+    MongoClientSettings.Builder settings =
+        MongoClientSettings.builder()
+            .clusterSettings(
+                ClusterSettings.builder()
+                    .description("some-description")
+                    .applyConnectionString(new ConnectionString("mongodb://" + host + ":" + port))
+                    .build());
+    settings.build();
+    MongoDatabase db = MongoClients.create(settings.build()).getDatabase(dbName);
+    db.createCollection(collectionName, toCallback(result -> {}));
+  }
+
+  @Override
+  public int getCollection(String dbName, String collectionName) {
+    MongoDatabase db = client.getDatabase(dbName);
+    CompletableFuture<Long> count = new CompletableFuture<>();
+    db.getCollection(collectionName).count(toCallback(o -> count.complete(((Long) o))));
+    return count.join().intValue();
+  }
+
+  @Override
+  public MongoCollection<Document> setupInsert(String dbName, String collectionName)
+      throws InterruptedException {
+    MongoCollection<Document> collection =
+        testing()
+            .runWithSpan(
+                "setup",
+                () -> {
+                  MongoDatabase db = client.getDatabase(dbName);
+                  CountDownLatch latch = new CountDownLatch(1);
+                  db.createCollection(collectionName, toCallback(result -> latch.countDown()));
+                  latch.await();
+                  return db.getCollection(collectionName);
+                });
+    ignoreTracesAndClear(1);
+    return collection;
+  }
+
+  @Override
+  public int insert(MongoCollection<Document> collection) {
+    CompletableFuture<Long> count = new CompletableFuture<>();
+    collection.insertOne(
+        new Document("password", "SECRET"),
+        toCallback(result -> collection.count(toCallback(o -> count.complete(((Long) o))))));
+    return count.join().intValue();
```

Comment:
did you choose to use `join` over `get` that was in the groovy version to not have to deal with the checked exceptions?

Reply:
Modified.